### PR TITLE
feat: index miners count and total rewards from fees

### PR DESCRIPTION
**Pull request by sborrazas**
refs #844 

Includes a migration that takes 15-30mins to index the already processed generation miners.

### CODE REVIEW

**Review by thepiwo — APPROVED** [review]

File: lib/ae_mdw/db/mutations/miner_rewards_mutation.ex [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            {State.put(state, Model.Miner, Model.miner(miner, total_reward: old_reward + reward)),
+             false}
+
+          :not_found ->
+            {State.put(
+               state,
+               Model.Miner,
+               Model.miner(index: beneficiary_pk, total_reward: reward)
+             ), true}
+        end
```
Comment (jyeshe) [inline lib/ae_mdw/db/mutations/miner_rewards_mutation.ex:36]: It's already readable but might convey the idea at a glance by giving a name to this `case` block with a function (sth like `increment_total_reward`)

File: lib/ae_mdw/db/mutations/miner_rewards_mutation.ex [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+      if new_miner? do
+        State.update(state, Model.Stat, Stats.miners_count_key(), fn
+          Model.stat(payload: count) = stat ->
+            Model.stat(stat, payload: count + 1)
+
+          nil ->
+            Model.stat(index: Stats.miners_count_key(), payload: 1)
+        end)
```
Comment (jyeshe) [inline lib/ae_mdw/db/mutations/miner_rewards_mutation.ex:45]: same here with `increment_payload`

File: lib/ae_mdw/miners.ex [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    {serialize_cursor(prev_cursor), Enum.map(miners, &fetch_miner!(state, &1)),
+     serialize_cursor(next_cursor)}
+  end
+
+  @spec fetch_miner!(state(), pubkey()) :: miner()
+  def fetch_miner!(state, miner_pk),
+    do: render_miner(State.fetch!(state, Model.Miner, miner_pk))
+
+  defp build_streamer(state, cursor), do: &Collection.stream(state, Model.Miner, &1, nil, cursor)
```
Comment (jyeshe) [inline lib/ae_mdw/miners.ex:37]: :+1:

File: priv/migrations/20220816120000_add_miners.ex [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  require Model
+
+  @log_freq 1_000
+
+  @spec run(boolean()) :: {:ok, {non_neg_integer(), non_neg_integer()}}
+  def run(_from_start?) do
+    state = State.new()
+
+    case state |> Collection.stream(Model.DeltaStat, :backward, nil, nil) |> Enum.take(1) do
```
Comment (jyeshe) [inline priv/migrations/20220816120000_add_miners.ex:20]: maybe `Database.last_key`

**Review by jyeshe — COMMENTED** [review]
No operational issues, only refactoring opportunities.

File: lib/ae_mdw/db/mutations/miner_rewards_mutation.ex (hunk 1 above)
Comment (sborrazas) [reply lib/ae_mdw/db/mutations/miner_rewards_mutation.ex]: Done

File: lib/ae_mdw/db/mutations/miner_rewards_mutation.ex (hunk 2 above)
Comment (sborrazas) [reply lib/ae_mdw/db/mutations/miner_rewards_mutation.ex]: Done

File: priv/migrations/20220816120000_add_miners.ex (hunk 4 above)
Comment (sborrazas) [reply priv/migrations/20220816120000_add_miners.ex]: Done

**Review by jyeshe — APPROVED** [review]

**Pull request merged**